lib/refs_updater.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+# Starts analysis jobs for all commits that were added since the old revision
+# for each ref.
+class RefsUpdater
+  attr_reader :user, :repository, :git, :updated_refs
+  def initialize(user, repository, updated_refs)
+    @user = user
+    @repository = RepositoryCompound.wrap(repository)
+    @git = @repository.git
+    @updated_refs = updated_refs
+  end
+
+  def call
+    updated_refs.each do |info|
+      analyze(info['ref'], info['before'], info['after'])
+    end
+  end
+
+  protected
+
+  def analyze(_ref, before, after)
+    # This assumes that it was _not_ a force-push. It can _only_ be done because
+    # force-pushes are prevented. With a force-push, we would need to find the
+    # first common ancestor C of `before` and `after`, analyze all commits from
+    # C onwards and delete `FileVersion`s after C leading to `before`.
```

Comment:
Unless these commits are also part of another branch

Reply:
Fixed with 2c2de71f

Reply:
Sorry, I meant that FileVersions for commits that are also part of another branch should not be deleted then.

Reply:
Fixed with 923c50b2. Note that I added a new commit (87ba412d) right before that one.